src/ucp/core/ucp_worker.c

```diff
@@ -3090,38 +3089,48 @@ void ucp_worker_keepalive_remove_ep(ucp_ep_h ep)
         return;
     }
 
-    round_inprogress = worker->keepalive.iter != worker->keepalive.iter_end;
+    if ((worker->keepalive.iter == &ucp_ep_ext_gen(ep)->ep_list) &&
+        (worker->keepalive.iter_end == &ucp_ep_ext_gen(ep)->ep_list)) {
```

Comment:
`/* we are removing the current and last ep in the keepalive round */`

Reply:
done